staging/script.xsqueeze/resources/lib/classes/NowPlayingWindow.py

```diff
@@ -295,7 +302,41 @@ def update(self):
         self.updatePlaylistDetails()
         self.updateCoverArtFromURLs()
 
-  ##############################################################################
+        #trigger a hdmi command, if a controller has changed the power status or the volume
+        if constants.TURNONAVR == 'true' and constants.CECSUPPORT == 'true':
+          powerstate = self.player.getpowerstate()
+          if powerstate == 0 and constants.HDMIPOWER == "true":
+            if constants.TURNOFFTV == "true":
+               tv.standby()
+               pass
+            if constants.TURNONAVR == "true":
+               avr.standby()
+            constants.HDMIPOWER = "false"
+          elif powerstate != 0 and constants.HDMIPOWER == "false":
+            if constants.TURNONAVR == "true":
+               avr.power_on()
+               avr.set_av_input(int(constants.AVRINPUT))
+            constants.HDMIPOWER = "true"
+          else:
+            pass
+
+        if constants.ADJUSTVOL == "true" and constants.CECSUPPORT == 'true':
+          vol1 = self.player.getVolume()
+          xbmc.sleep(1000)
+          vol2 = self.player.getVolume()
+          counter = 0 - int(constants.VOLFACTOR)
+          if vol2 > vol1:
+            while counter <= abs(vol1 -vol2):
+              cec.volume_up()
+              counter = counter + 1
+          elif vol2 < vol1:
+            while counter <= abs(vol1 -vol2) :
+              cec.volume_down()
+              counter = counter + 1
+          elif vol2 == vol1:
+            pass
+
+##############################################################################
   # Kick off the artist.slideshow
```

Comment:
I think you should move the above into it's own function and call it only if there is ce support.  Don't muddy the main loop with a lot of task specific code, basically.